[2024-11-23T18:13:47.876+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-23T18:13:47.967+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: fraud_detection.process_transaction scheduled__2024-11-23T18:12:00+00:00 [queued]>
[2024-11-23T18:13:48.020+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: fraud_detection.process_transaction scheduled__2024-11-23T18:12:00+00:00 [queued]>
[2024-11-23T18:13:48.021+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 2
[2024-11-23T18:13:48.106+0000] {taskinstance.py:2888} INFO - Executing <Task(BranchPythonOperator): process_transaction> on 2024-11-23 18:12:00+00:00
[2024-11-23T18:13:48.135+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=4892) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-11-23T18:13:48.137+0000] {standard_task_runner.py:72} INFO - Started process 4897 to run task
[2024-11-23T18:13:48.143+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'fraud_detection', 'process_transaction', 'scheduled__2024-11-23T18:12:00+00:00', '--job-id', '487', '--raw', '--subdir', 'DAGS_FOLDER/fraud_detection_dag.py', '--cfg-path', '/tmp/tmpqoye5nic']
[2024-11-23T18:13:48.146+0000] {standard_task_runner.py:105} INFO - Job 487: Subtask process_transaction
[2024-11-23T18:13:48.277+0000] {task_command.py:467} INFO - Running <TaskInstance: fraud_detection.process_transaction scheduled__2024-11-23T18:12:00+00:00 [running]> on host 345e92ff26bf
[2024-11-23T18:13:48.638+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='fraud_team' AIRFLOW_CTX_DAG_ID='fraud_detection' AIRFLOW_CTX_TASK_ID='process_transaction' AIRFLOW_CTX_EXECUTION_DATE='2024-11-23T18:12:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-11-23T18:12:00+00:00'
[2024-11-23T18:13:48.671+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-23T18:13:48.734+0000] {baseoperator.py:405} WARNING - BranchPythonOperator.execute cannot be called outside TaskInstance!
[2024-11-23T18:13:48.774+0000] {fraud_detection_dag.py:78} INFO - Données récupérées depuis XCom: {"columns":["cc_num","merchant","category","amt","first","last","gender","street","city","state","zip","lat","long","city_pop","job","dob","trans_num","merch_lat","merch_long","is_fraud","current_time"],"index":[134940],"data":[[3595192916105588,"fraud_Waters-Cruickshank","health_fitness",55.58,"Dustin","Roberts","M","3283 James Station","Fayetteville","NC",28314,35.0583,-79.008,238602,"Scientist, research (maths)","1993-05-14","4ee9de8341fbcf16c29da39f8d69c1cc",35.498726,-79.167346,0,1732385622810]]}
[2024-11-23T18:13:48.775+0000] {fraud_detection_dag.py:86} INFO - Données JSON parsées avec succès
[2024-11-23T18:13:48.785+0000] {fraud_detection_dag.py:90} INFO - DataFrame créé avec 1 lignes
[2024-11-23T18:13:48.790+0000] {fraud_detection_dag.py:94} INFO - Colonnes renommées avec succès
[2024-11-23T18:13:48.791+0000] {fraud_detection_dag.py:95} INFO - Colonnes disponibles: ['cc_num', 'merchant', 'category', 'amt', 'first', 'last', 'gender', 'street', 'city', 'state', 'zip', 'lat', 'long', 'city_pop', 'job', 'dob', 'trans_num', 'merch_lat', 'merch_long', 'is_fraud', 'trans_date_trans_time']
[2024-11-23T18:13:48.794+0000] {fraud_detection_dag.py:99} INFO - Conversion de trans_date_trans_time en unix_time réussie
[2024-11-23T18:13:48.867+0000] {fraud_detection_dag.py:108} INFO - Transformation des données...
[2024-11-23T18:13:50.705+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/sklearn/ensemble/_base.py:172: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = min(effective_n_jobs(n_jobs), n_estimators)

[2024-11-23T18:13:50.767+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/sklearn/ensemble/_base.py:172: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = min(effective_n_jobs(n_jobs), n_estimators)

[2024-11-23T18:13:50.781+0000] {fraud_detection_dag.py:120} INFO - Prediction de fraude: NORMAL, Probabilité: 30.56%
[2024-11-23T18:13:50.910+0000] {fraud_detection_dag.py:126} INFO - Prédiction: NORMAL (Probabilité: 30.56%)
[2024-11-23T18:13:50.911+0000] {python.py:240} INFO - Done. Returned value was: notify_normal
[2024-11-23T18:13:50.914+0000] {branch.py:38} INFO - Branch into notify_normal
[2024-11-23T18:13:50.919+0000] {skipmixin.py:230} INFO - Following branch ('notify_normal',)
[2024-11-23T18:13:50.967+0000] {skipmixin.py:278} INFO - Skipping tasks [('notify_fraud', -1), ('skip_processing', -1)]
[2024-11-23T18:13:51.076+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-23T18:13:51.078+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=fraud_detection, task_id=process_transaction, run_id=scheduled__2024-11-23T18:12:00+00:00, execution_date=20241123T181200, start_date=20241123T181347, end_date=20241123T181351
[2024-11-23T18:13:51.154+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2024-11-23T18:13:51.290+0000] {taskinstance.py:3900} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2024-11-23T18:13:51.321+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
